release/models/system/openconfig-gnsi-acctz.yang

```diff
@@ -0,0 +1,195 @@
+            enum GNOI {
+                value 2;
+            }
+            enum GNSI {
+                value 3;
+            }
+            enum GRIBI {
+                value 4;
+            }
+            enum P4RT {
+                value 5;
+            }
+        }
+    }
+    typedef service-request {
+        description "enum RecordResponse.service_request";
+        type enumeration {
+            enum cmd_service {
+                value 4;
+            }
+            enum grpc_service {
+                value 5;
+            }
+        }
+    }
+    typedef service-type {
+        description "enum cmd or grpc service type";
+        type union {
+            type cmd_service;
+            type grpc_service;
+        }
+    }
+
+    // gnsi.acctz client statistics
+    grouping client-counters {
+        description
+          "A collection of counters that were collected by the gNSI.acctz
+           module while servicing acctz clients.";
+
+        leaf history_istruncated {
+            type oc-yang:counter64;
+            description
+              "The total number of times that a RecordRequest resulted in
+               a RecordResponse being marked history_istruncated.  ie: a
+               request was made for a timestamp that did not exist in the
+               history.";
+        }
+        leaf IdleTimeouts {
+            type oc-yang:counter64;
+            description
+              "The total number of times that a client was disconnected
+               due to missing keepalives (ie: RecordRequests).";
+        }
+        leaf RecordRequests {
+            type oc-yang:counter64;
+            description
+              "The total number of RecordRequest RPCs that have been received.";
+        }
+        leaf RecordResponses {
+            type oc-yang:counter64;
+            description
+              "The total number of RecordRequest RPCs that have been sent.";
+        }
+    }
+
+    // gnsi.acctz producer statistics
+    grouping source-counters {
+        description
+          "A collection of counters for gNSI.acctz record production per
+           service request type.";
+
+        list source-records {
+            key "service type";
+            // unique "service type";
+            description
+              "The total number of times the gNSI.authz module denied access
+               to a RPC.";
+
+            leaf service {
+                type service-request;
+                mandatory true;
+            }
+            leaf type {
+                type service-type;
+                mandatory true;
+            }
+        leaf records {
+            type oc-yang:counter64;
+            description
+              "The total number of records produced for the service_request
+               type.";
+        }
+        }
+    }
+
+    grouping grpc-server-acctz-counters {
+        description
+          "A collection of counters from the gNSI.acctz module.";
+
+        container counters {
```

Comment:
Thanks for the suggestion. I've added `identity GNSI` which can be specified as a leaf-list element at https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-system.html#system-grpc-servers-grpc-server-config-services

Do you happen to know what the `when` statement's XPATH expression would look like for checking whether a leaf-list contains an element?

Reply:
Thx, I had started on a refactor of the grpc-server model to accommodate more fine grained service related parameters that I'll resurrect as a separate PR and one of the items in here I was contemplating was if we want to use this leaf-list of identities as the enablement or disablement of a group of services or merely its use as "This gRPC server hosts these services" where services each have containers (gated by existence of this leaf-list entry) w/ a subsequent boolean enable node (much like we have throughout the rest of the model set)

Then within each service category - we have any other related config/state specific to that service (where much of this state can get relocated)

This would essentially be a `when` statement on each augment in gNSI modules to mount to a location within a `/system` subtree _if_ there is existence of a leaf-list value of that equivalent service

e.g.
```
          container gnmi {
            when "../../state/services = 'oc:GNMI'";
          }
          container gnoi {
            when "../../state/services = 'oc:GNOI'";
          }
          container gnsi {
            when "../../state/services = 'oc:GNSI'";
          }
        }
```

Gimme a few days and I'll draft up the minimum `grpc-server` modifications we would want here to accommodate this

Reply:
I think this totally makes sense. I'll let you make the separate PR on that.

Meanwhile I'm using
`when "../../state/services[contains(., 'oc-sys-grpc:GNMI')] = 'oc-sys-grpc:GNMI'";`
for the when statement, pyang doesn't complain, so hopefully it's correct.

Reply:
These are the statements currently being used:
```yang
when "../config[contains(services, 'oc-gnsi:GNSI')]/enable = 'true'";
```